### ISSUE TITLE
Is there any recommendations on number of topics per Consumer instance?

### ISSUE THREAD

**Issue by alex-basiuk** [confluentinc/confluent-kafka-dotnet#1286]
I've got an app that should consume N topics where N is relatively small (up to 5). Basically, I have 2 options:
1) **1 to 1 relationship**: Create a consumer instance which is subscribed to only one topic. So in total I will have N consumer instances per process.
2) **1 to N relationship**: Create a single consumer instances per process which is subscribed to all N topics.

I'm leaning towards option 1) because it simplifies user code. Option 2) would require some sort of custom routing to the respective message handlers.

Is there any recommendations which option is preferable?

**Comment by Rytisk** [comment]
Check this issue: #197 

It mentions that to maximize throughput, it's better to have a single consumer instance per process. Since with multiple consumer, you duplicate connections to broker, add some overhead with multiple TCP requests and prevent the broker to batch all the data it can.

**Comment by mhowlett** [comment]
yep, what @Rytisk is right. but as you say, the flip side of that it can complicate the code (sounds like it does for your use case). 

if efficiency is not a big concern (your throughput is relatively low by kafka standards, and the protocol level bandwidth overhead per consumer isn't significant for you), then there's a lot to be said for simple code.

I think we should add a app level routing api on top of the basic consumer, and furthermore, this should also be capable of routing based on schema (so you can have multiple schemas in a given topic). if you go with a single consumer, you should probably make that abstraction yourself.

the current API reflects Kafka's original use case, which is massive throughput, but relatively simple routing.
